viewing_party/party.py

```diff
@@ -1,23 +1,200 @@
+        if not friends_movies_list[movie]["title"] in user_data_movies_titles_list:
+            if friends_movies_list[movie] in friends_unique_movies_list:
+                continue
+            else:
+                friends_unique_movies_list.append(friends_movies_list[movie])
+
+    return friends_unique_movies_list
+
 
-        
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
+def get_available_recs(user_data):
+    subscriptions = user_data["subscriptions"]
+    
+    friends_unique_movies_list = get_friends_unique_watched(user_data)
+
+    movie_recommendations_list = []
+
+    for movie in range(len(friends_unique_movies_list)):
+        if friends_unique_movies_list[movie]["host"] in subscriptions:
+            movie_recommendations_list.append(friends_unique_movies_list[movie])
+
+    return movie_recommendations_list
 
 # -----------------------------------------
 # ------------- WAVE 5 --------------------
 # -----------------------------------------
+def get_new_rec_by_genre(user_data):  
+    user_data_movies_list = get_user_data_watchlist(user_data)
+
+    friends_unique_movies_list = get_friends_unique_watched(user_data)
+    
+    genre_list = []
+    for movie in range(len(user_data_movies_list)):
+        genre = user_data_movies_list[movie]["genre"]
+        genre_list.append(genre)
+    
+    movie_recommendations_list = []
+    if len(user_data["watched"]) > 0:
+        most_watched_genre = max(set(genre_list), key = genre_list.count)
+
+        for movie in range(len(friends_unique_movies_list)):
+            if friends_unique_movies_list[movie]["genre"] == most_watched_genre:
+                movie_recommendations_list.append(friends_unique_movies_list[movie])
+
+    return movie_recommendations_list
+
+
+def get_rec_from_favorites(user_data):
+    unique_user_data_movies_list = get_unique_watched(user_data)
+    
+    favorites = user_data["favorites"]
+    
+    movie_recommendations_list = []
+    if len(user_data["watched"]) > 0:
+        for movie in range(len(unique_user_data_movies_list)):
+            if unique_user_data_movies_list[movie] in favorites:
+                movie_recommendations_list.append(unique_user_data_movies_list[movie])
+
+    return movie_recommendations_list
+
+
+# -----------------------------------------
+# ------------- HELPERS -------------------
+# -----------------------------------------
+def get_user_data_watchlist(user_data):
+    user_data_movies_list = []
+
+    for i in range(len(user_data["watched"])):
+        user_data_movies_list.append(user_data["watched"][i])
+
+    return user_data_movies_list
+
+def get_user_data_watchlist_titles(user_data):
+    user_data_movies_titles_list = []
+
+    for i in range(len(user_data["watched"])):
+        user_data_movies_titles_list.append(user_data["watched"][i]["title"])
+
+    return user_data_movies_titles_list

```

Comment:
Once again, moving through the titles works, but it's a bit more streamlined if we go ahead and work with the entire movies themselves!